loader.c

```diff
@@ -309,7 +308,15 @@ dloader_p api_load(const char *filename)
         case R_ARM_RELATIVE:
         {
             ElfW(Addr) *addr = (ElfW(Addr) *)(load_bias + reloc->r_offset);
-            *addr += load_bias;
+            if ((intptr_t) addr < ro_end && (intptr_t) addr >= ro_start) {
```

Comment:
The code looks tricky to me. Please add comments around.

Reply:
Done. Review please.